commit 17206cc8c21c439d121a66d7c9934cdfa4791a35
Author: Mark Thomas <markt@apache.org>
Date:   Wed Sep 16 13:37:35 2015 +0000

    Fix https://bz.apache.org/bugzilla/show_bug.cgi?id=58386
    On the basis that access() and finish() are synchronized, extend synchronization to other methods that access same fields.
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1703408 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/catalina/tribes/io/ObjectReader.java b/java/org/apache/catalina/tribes/io/ObjectReader.java
index 6ea307e..9d1b78a 100644
--- a/java/org/apache/catalina/tribes/io/ObjectReader.java
+++ b/java/org/apache/catalina/tribes/io/ObjectReader.java
@@ -82,7 +82,7 @@ public class ObjectReader {
         this.lastAccess = System.currentTimeMillis();
     }
 
-    public boolean isAccessed() {
+    public synchronized boolean isAccessed() {
         return this.accessed;
     }
 
@@ -150,7 +150,7 @@ public class ObjectReader {
         this.buffer = null;
     }
 
-    public long getLastAccess() {
+    public synchronized long getLastAccess() {
         return lastAccess;
     }
 
@@ -158,7 +158,7 @@ public class ObjectReader {
         return cancelled;
     }
 
-    public void setLastAccess(long lastAccess) {
+    public synchronized void setLastAccess(long lastAccess) {
         this.lastAccess = lastAccess;
     }
 
